### PR TITLE
Update test_interfaces to include new interface modules.

### PR DESCRIPTION
**Pull request by lance-wilson**
# Reviewer Checklist
https://github.com/NRLMMD-GEOIPS/.github/blob/main/.github/review-template.md

# Related Issues
fixes NRLMMD-GEOIPS/geoips#91
fixes NRLMMD-GEOIPS/geoips#90  <-- Found and fixed this bug while I was in here.

# Testing Instructions

Run test_interfaces.py in geoips/commandline.

# Summary
## GEOIPS/geoips#91: 2023-02-14, update test_interfaces

*From issue NRLMMD-GEOIPS/geoips#91: 2023-02-14, update test_interfaces*
************************************************************************

Bug fixes
=========

Removed unused import of Hashable
---------------------------------

*From issue NRLMMD-GEOIPS/geoips#91: 2023-02-14, update test_interfaces*

::

    geoips/interface_modules/readers/abi_netcdf.py
    geoips/interface_modules/readers/ahi_hsd.py

Switched coverage from positional parameter to keyword argument
---------------------------------------------------------------

Standard filename format expects "coverage" to be a keyword argument, not a positional
parameter.  Update filename formats accordingly so test_interfaces passes.

::

    modified: geoips/interface_modules/filename_formats/geoips_fname.py
    modified: geoips/interface_modules/filename_formats/geotiff_fname.py
    modified: geoips/interface_modules/filename_formats/tc_clean_fname.py
    modified: geoips/interface_modules/filename_formats/tc_fname.py

Installation and Test
=====================

Add validation tests to Base interface class
--------------------------------------------

*From issue NRLMMD-GEOIPS/geoips#91: 2023-02-14, update test_interfaces*

* plugin_is_valid: Added checks for required arguments and keyword arguments
  to determine if a plugin call signature matches what is expected
* get_plugin, get_plugins, test_interface_plugins: A few bug fixes, but mostly
  changes to avoid future functionality to get tests to run successfully
* plugins_all_valid: returns True if all plugins in the current interface are valid,
  False if any plugins are invalid (calls "plugin_is_valid" on each plugin)

::

    modified:   geoips/interfaces/base.py

Add separate interface tests for deprecated and new interfaces
--------------------------------------------------------------

* Move tests for deprecated interfaces to a separate function - will eventually be
  removed.
* Call both "plugins_all_valid" and "test_interface_plugins" for new interfaces.

  * "plugins_all_valid" merely returns True or False for full interface (if any fail,
    returns False, error raised at the end of "test_interfaces")
  * "test_interface_plugins" actually opens every plugin, and tests individually - if
    any fail, they will be added to the failed_plugins list and an error raised at the
    end for non-zero return.

* Add logic to save lists of successful and failed interfaces - print all status at
  the end, and only raise an error after printing full list of successful/failed
  plugins.

::

    modified:   geoips/commandline/test_interfaces.py

Update interface modules for validation testing
-----------------------------------------------

*From issue NRLMMD-GEOIPS/geoips#91: 2023-02-14, update test_interfaces*

* Add required_args and required_kwargs to all interface modules
* Standardize interface class names
  * FilenameFormats -> FilenameFormatsInterface
  * ColorMapInterface -> ColormapsInterface
  * TitleFormattersInterface -> TitleFormatsInterface

::

    modified:   geoips/interfaces/algorithms.py
    modified:   geoips/interfaces/colormaps.py
    modified:   geoips/interfaces/filename_formats.py
    modified:   geoips/interfaces/interpolators.py
    modified:   geoips/interfaces/output_formats.py
    modified:   geoips/interfaces/procflows.py
    modified:   geoips/interfaces/readers.py
    modified:   geoips/interfaces/title_formats.py

# Output

Output of test_interfaces:
Mon Feb 13 16:54:16 CST 2023  Running  test_interfaces
test_interfaces
/home/lwilson/geoproc/test_data//logs/20230213/20230213.225343_geoips_base/test_all_geoips_base.log_test_interfaces.log
Return: 0 
[test_all_geoips_base.log_test_interfaces.log](https://github.com/NRLMMD-GEOIPS/geoips/files/10736340/test_all_geoips_base.log_test_interfaces.log)

Output images of test_base_install_low_memory.sh:
20200918 195020 goes-16 abi Infrared goes16 45p56 noaa 10p0
20200918_195020_AL202020_abi_goes-16_Infrared_110kts_100p00_1p0

### CODE REVIEW

**Comment by mindyls** [comment]
I made some changes here, mostly to:
1. Separate out the "old" and new interface testing in test_interfaces (since we'll just delete that whole section once the YAML plugins are finished)
2. Added support for checking kwargs
3. Updated test_interfaces so it goes through all the plugins (rather than just failing as soon as it gets to a bad one), and prints out a list of all the successful/failed plugins at the end.

In the process, I found I was not fully testing things before, and I had to update some of the filename_format plugins to comply.

I think this is good to go now.  @jsolbrig @lance-wilson  let me know if you have any additional comments / suggestions.

```
SUCCESSFUL PLUGIN filename_formats on text_winds_tc_fname
SUCCESSFUL PLUGIN interpolators on interp_gauss
SUCCESSFUL PLUGIN interpolators on interp_nearest
SUCCESSFUL PLUGIN interpolators on interp_grid
SUCCESSFUL PLUGIN output_formats on full_disk_image
SUCCESSFUL PLUGIN output_formats on imagery_annotated
SUCCESSFUL PLUGIN output_formats on imagery_windbarbs
SUCCESSFUL PLUGIN output_formats on geotiff_standard
SUCCESSFUL PLUGIN output_formats on imagery_clean
SUCCESSFUL PLUGIN output_formats on imagery_windbarbs_clean
SUCCESSFUL PLUGIN output_formats on metadata_default
SUCCESSFUL PLUGIN output_formats on metadata_tc
SUCCESSFUL PLUGIN output_formats on netcdf_geoips
SUCCESSFUL PLUGIN output_formats on netcdf_xarray
SUCCESSFUL PLUGIN output_formats on text_winds
SUCCESSFUL PLUGIN output_formats on unprojected_image
SUCCESSFUL PLUGIN procflows on config_based
SUCCESSFUL PLUGIN procflows on single_source
SUCCESSFUL PLUGIN readers on abi_l2_netcdf
SUCCESSFUL PLUGIN readers on abi_netcdf
SUCCESSFUL PLUGIN readers on ahi_hsd
SUCCESSFUL PLUGIN readers on amsr2_netcdf
SUCCESSFUL PLUGIN readers on amsr2_remss_winds_netcdf
SUCCESSFUL PLUGIN readers on amsub_hdf
SUCCESSFUL PLUGIN readers on amsub_mirs
SUCCESSFUL PLUGIN readers on ascat_uhr_netcdf
SUCCESSFUL PLUGIN readers on atms_hdf5
SUCCESSFUL PLUGIN readers on ewsg_netcdf
SUCCESSFUL PLUGIN readers on geoips_netcdf
SUCCESSFUL PLUGIN readers on gmi_hdf5
SUCCESSFUL PLUGIN readers on imerg_hdf5
SUCCESSFUL PLUGIN readers on mimic_netcdf
SUCCESSFUL PLUGIN readers on modis_hdf4
SUCCESSFUL PLUGIN readers on saphir_hdf5
SUCCESSFUL PLUGIN readers on sar_winds_netcdf
SUCCESSFUL PLUGIN readers on scat_knmi_winds_netcdf
SUCCESSFUL PLUGIN readers on seviri_hrit
SUCCESSFUL PLUGIN readers on sfc_winds_text
SUCCESSFUL PLUGIN readers on smap_remss_winds_netcdf
SUCCESSFUL PLUGIN readers on smos_winds_netcdf
SUCCESSFUL PLUGIN readers on ssmi_binary
SUCCESSFUL PLUGIN readers on ssmis_binary
SUCCESSFUL PLUGIN readers on viirs_netcdf
SUCCESSFUL PLUGIN readers on wfabba_ascii
SUCCESSFUL PLUGIN readers on windsat_idr37_binary
SUCCESSFUL PLUGIN readers on windsat_remss_winds_netcdf
SUCCESSFUL PLUGIN readers on clavrx_hdf4
SUCCESSFUL PLUGIN title_formats on static_standard
SUCCESSFUL PLUGIN title_formats on tc_copyright
SUCCESSFUL PLUGIN title_formats on tc_standard
SUCCESSFUL INTERFACE algorithms
SUCCESSFUL INTERFACE colormaps
SUCCESSFUL INTERFACE filename_formats
SUCCESSFUL INTERFACE interpolators
SUCCESSFUL INTERFACE output_formats
SUCCESSFUL INTERFACE procflows
SUCCESSFUL INTERFACE readers
SUCCESSFUL INTERFACE title_formats
```

Output with abi_netcdf reader slightly modified:
```
-def abi_netcdf(
-    fnames, metadata_only=False, chans=None, area_def=None, self_register=False
-):
+def abi_netcdf(fnmes, metadata_only, chans=None, area_def=None, self_register=False):
     """
     Read ABI NetCDF data from a list of filenames.

SUCCESSFUL INTERFACE algorithms
SUCCESSFUL INTERFACE colormaps
SUCCESSFUL INTERFACE filename_formats
SUCCESSFUL INTERFACE interpolators
SUCCESSFUL INTERFACE output_formats
SUCCESSFUL INTERFACE procflows
SUCCESSFUL INTERFACE title_formats
FAILED PLUGIN readers on abi_netcdf
FAILED INTERFACE readers
Traceback (most recent call last):
  File "$GEOIPS_BASEDIR/geoips_dependencies/miniconda3/envs/geoips_conda/bin/test_interfaces", line 33, in <module>
    sys.exit(load_entry_point('geoips', 'console_scripts', 'test_interfaces')())
  File "$GEOIPS/geoips/commandline/test_interfaces.py", line 158, in main
    raise TypeError(f"Failed validity check on plugins {failed_plugins}")
TypeError: Failed validity check on plugins ['readers on abi_netcdf']
(geoips_conda) echo $?
1
```

**Review by mindyls — DISMISSED** [review]
This looks good to me, but considering I made the final set of changes, not sure my approval really counts...

But, if someone else agrees, feel free to merge.

**Review by mindyls — APPROVED** [review]
This looks good to me, but considering I made the final set of changes (plus the merge conflicts I just resolved), not sure my approval really counts...

But, if someone else agrees, feel free to merge.

File: geoips/commandline/test_interfaces.py [hunk 1]
```diff
@@ -53,21 +51,107 @@ def main():
...
+    interfaces = [
+        "algorithms",
+        "colormaps",
+        "filename_formats",
+        "interpolators",
+        "output_formats",
+        "procflows",
+        "readers",
+        "title_formats",
+    ]
```
Comment (jsolbrig) [inline geoips/commandline/test_interfaces.py:103]:
```suggestion
    interfaces = [
        "algorithms",
        "colormaps",
        "filename_formats",
        "interpolators",
        "output_formats",
        "procflows",
        "readers",
        "title_formats",
    ]
```

Should this just be determined by what exists in `interfaces`? Something like:
```
interfaces = [inter for inter in geoips.interfaces if isinstance(inter, geoips.interfaces.BaseInterface) and type(inter) not geoips.interfaces.BaseInterface]
```

If so, I'll make the change. There are a couple little details to take care of here.

Comment (mindyls) [reply geoips/commandline/test_interfaces.py:103]: Sure, you can make that change

Comment (jsolbrig) [reply geoips/commandline/test_interfaces.py:103]: I will resolve this in a separate PR. This is more involved than I thought.

**Review by jsolbrig — APPROVED** [review]
I'll make the changes that I had suggested in a separate PR. They're more involved than I thought. This looks good to me.

**Pull request merged**